src/dbnode/persist/fs/index_read.go

```diff
@@ -248,6 +248,12 @@ func (r *indexReader) ReadSegmentFileSet() (
 			segmentFileType: segFileType,
 			digest:          digest.Checksum(bytes),
 		})
+
+		// NB(bodu): Free mmaped bytes after we take the checksum so we don't get memory spikes at bootstrap time.
+		if err := mmap.Free(bytes); err != nil {
+			closeFiles()
```

Comment:
nit: I know this is not your original code, but maybe best to use the following pattern that we do elsewhere to always call `closeFiles()` in case of error:
```go
success := false
defer func() {
  if success {
    return // Returned successfully, do not need to close files
  }
  for _, file := range result.files { // This was originally closeFiles()
    file.Close()
  }
}()


// Rest of the code here

// Right before return successfully
success = true
return result, nil
```

Reply:
Then you don't need to call `closeFiles()` here explicitly and any new code paths automatically get the same behavior for free.